workflows/funcscan.nf

```diff
@@ -155,23 +156,34 @@ workflow FUNCSCAN {
     */
 
     // Some tools require annotated FASTAs
-    // For prodigal run twice, once for gff and once for gbk generation, (for parity with PROKKA which produces both)
+    // For prodigal: run twice, once for gff and once for gbk generation, (for parity with PROKKA which produces both)
     if ( ( params.run_arg_screening && !params.arg_skip_deeparg ) || ( params.run_amp_screening && ( !params.amp_skip_hmmsearch || !params.amp_skip_amplify || !params.amp_skip_ampir ) ) || ( params.run_bgc_screening && ( !params.bgc_skip_hmmsearch || !params.bgc_skip_antismash ) ) ) {
 
         if ( params.annotation_tool == "prodigal" ) {
             PRODIGAL_GFF ( ch_prepped_input, "gff" )
+            GUNZIP_FAA ( PRODIGAL_GFF.out.amino_acid_fasta )
+            GUNZIP_FNA ( PRODIGAL_GFF.out.nucleotide_fasta)
+            GUNZIP_GFF ( PRODIGAL_GFF.out.gene_annotations )
             ch_versions              = ch_versions.mix(PRODIGAL_GFF.out.versions)
-            ch_annotation_faa        = PRODIGAL_GFF.out.amino_acid_fasta
-            ch_annotation_fna        = PRODIGAL_GFF.out.nucleotide_fasta
-            ch_annotation_gff        = PRODIGAL_GFF.out.gene_annotations
-            ch_annotation_gbk        = Channel.empty() // Prodigal doesn't produce GBK
+            ch_annotation_faa        = GUNZIP_FAA.out.gunzip
+            ch_annotation_fna        = GUNZIP_FNA.out.gunzip
+            ch_annotation_gff        = GUNZIP_GFF.out.gunzip
+            ch_annotation_gbk        = Channel.empty() // Prodigal GBK and GFF output are mutually exclusive
```

Comment:
When I see this, I feel it might be better to allow modules to read gzipped files, at least by unzipping if necessary. Would mean a few fewer process calls, if no disk space saved (although a module could, I suppose, remove a gunzipped file after a successful run; I've never seen modules do that). `gunzip` is available in containers, I believe.

Reply:
So it would be a trade-off between several module updates or having the few gunzip lines in this workflow. Since I would keep the module behaviour as close to the original tools as possible (i.e. modules of tools which don't accept zipped input, should also not accept it), I would keep it as is.
Other thoughts, @jfy133 ?

Reply:
Yes, the vast majority of tools in the pipeline _do not_ support gzipped files unfortunately natively. So in fact, having a single gzip module that then sends everythign downstream requires _less_ computation than unzipping in every single module, as in that case you would have to repeatedly unzip the same sample within each tool...

Limitation of the field unfortunaetly :(